### DISCUSSION TITLE
For my Gradio build, I've added frame interpolation tweening.

### DISCUSSION THREAD

**Discussion by ourladyofagi** [deforum-art/sd-webui-text2video#65] [General]
Which definitely helps to de-headache-ify the outputs. 

I also added an optical flow add/subtract/multiply/overlay/screen widget with some sliders, just because. Still messing with them. The frame interpolation is really easy to implement in CV2 but may exceed the scope/purpose of your build. 

(And when I say "I," of course I mean "me and ChatGPT4" lol.)

https://user-images.githubusercontent.com/126371399/227706203-8499a38a-6f3b-40ec-80aa-8fc2e62bdfc4.mp4

**Comment by hithereai** [comment]
Welcome to PR so we can consider it.

**Reply by ourladyofagi** [reply]
I'm just an abject hack who is working on a ridiculous on-brand cyberpunk-spoopy TikTok tentacle goth girl content generator haha. I would never presume that anything I suggest be useful to anyone but I figured I'd mention it anyways. :) <3


https://user-images.githubusercontent.com/126371399/227708230-863554f3-a535-4613-b8bb-b1a7128f3c5c.mp4

**Reply by hithereai** [reply]
You'd be surprised. Do you know how to PR? Need help?

**Reply by ourladyofagi** [reply]
I don't really use Github except to loot other people's stuff, but ChatGPT4 has made it so that I don't even have to do that. Anyways I just went to the Pull Requests for the first time, and I see that kabachuha and I had the same idea at the same time. GMTA. :)

I dropped a note to the DAMO ViLabs dude and he responded quickly.

<img width="432" alt="Screenshot 2023-03-25 023116" src="https://user-images.githubusercontent.com/126371399/227709110-f9921d19-0bcf-4af0-b582-6c3f92c19ae1.png">

EDIT: *my looting behavior is rooted in my programming incapacities, not malice or greed; my programming skills peaked with Turbo Pascal

**Comment by Cubey42** [comment]
> Which definitely helps to de-headache-ify the outputs.
> 
> I also added an optical flow add/subtract/multiply/overlay/screen widget with some sliders, just because. Still messing with them. The frame interpolation is really easy to implement in CV2 but may exceed the scope/purpose of your build.
> 
> (And when I say "I," of course I mean "me and ChatGPT4" lol.)

if you don't mind my asking, what settings did you do to produce this? I found alot of trouble with anything large becoming completely incoherent, can I ask for your parameters?

**Reply by ourladyofagi** [reply]
That one was 100 frames, 100 steps, with 30CFG, 4 tween frames and a bit of overlay with my optical flow widget that I'm not entirely sure does what I think it's doing haha. (The one thing my build is missing is the ability to set the seed, and neither I nor ChatGPT4 have figured out how to do it, so I'm unable to do more exacting tests.) 

I, too, am trying to get to the bottom of the coherency issue. 

https://user-images.githubusercontent.com/126371399/227743145-0a3c0655-50ac-4bb2-8c06-e759f6c33953.mp4

**Reply by Cubey42** [reply]
to achieve 384x768, what kinda of vram are you dealing with?

**Reply by ourladyofagi** [reply]
I'm generating them at 192x384 and just upscaling them. 

https://user-images.githubusercontent.com/126371399/227804393-c9573748-36b1-4df1-a31f-0bfb7fb80241.mp4

**Reply by ourladyofagi** [reply]
https://user-images.githubusercontent.com/126371399/227805625-1c01f3e3-f5c1-4993-b2db-4fcf9358bfdb.mp4

**Comment by ourladyofagi** [comment]
Temporal smoothing might be another thing worth checking out. Initial texts look good. Between that, the tweening, and a little bit of overlayblending in the alpha channel, it's nearly... yknow... good haha
(Sorry about spamming up your GitHub but I never last more than a day on reddit.)

https://user-images.githubusercontent.com/126371399/227756365-a50f4e47-1352-4656-85ad-f064b9fdb19a.mp4

**Reply by hithereai** [reply]
Join our Discord maybe? https://discord.gg/deforum
